core/src/com/unciv/logic/map/TileMap.kt

```diff
@@ -179,7 +179,7 @@ class TileMap {
 
                 val containsViewableNeighborThatCanSeeOver = tile.neighbors.any {
                     val neighborHeight = it.getHeight()
-                    viewableTiles.contains(it) && (
+                    !viewableTiles.contains(it) && (
```

Comment:
only if we don't have this tile already in `viewableTiles`

Reply:
No, this is asking "does this tile have a viewable neighbor"
So it needs to be contains()

Reply:
ok revert